.editorconfig

```diff
@@ -0,0 +1,22 @@
+# http://editorconfig.org
+
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+
+[BUILD]
+indent_size = 4
+
+[CMakeLists.*]
+indent_size = 2
```

Comment:
* `google-cloud-cpp/CMakeLists.txt` is mostly 2.
* `google-cloud-cpp/bigtable/CMakeLists.txt` is mostly 4.
* `google-cloud-cpp/cmake/*` is all 4.

Should we make this 4 and fix the remaining cases?  It seems more readable to me.

Reply:
I guess using 4 here would make CMake more compatible with future BUILD files, so SGTM, and since you say it's more readable, then let's go with that.

Reply:
Changed to 4 space indent; will merge shortly (Travis CI is disabled for this PR).